src/coreComponents/physicsSolvers/fluidFlow/CompositionalMultiphaseStatistics.cpp

```diff
@@ -307,51 +304,57 @@ void CompositionalMultiphaseStatistics::computeRegionStatistics( real64 const ti
                                         subRegionPhaseMass.toView(),
                                         subRegionTrappedPhaseMass.toView(),
                                         subRegionImmobilePhaseMass.toView(),
-                                        subRegionComponentMass.toView() );
+                                        subRegiondissolvedComponentMass.toView() );
 
     ElementRegionBase & region = elemManager.getRegion( ElementRegionBase::getParentRegion( subRegion ).getName() );
-    RegionStatistics & regionStatistics = region.getReference< RegionStatistics >( viewKeyStruct::regionStatisticsString() );
+    RegionStatistics & regionStatistics = region.getGroup< RegionStatistics >( viewKeyStruct::regionStatisticsString() );
 
-    regionStatistics.averagePressure += subRegionAvgPresNumerator;
-    if( subRegionMinPres < regionStatistics.minPressure )
+    real64 & averagePressure = regionStatistics.getReference< real64 >( RegionStatistics::viewKeyStruct::averagePressureString());
```

Comment:
I'm really not fan of passing by the `RegionStatistics` -> `wrappers` map -> wrapper entry by `viewKey` string -> back to reference to `RegionStatistics` value -> `RegionStatistics` actual value, to just read then write in a `RegionStatistics` value...
I think we should have some reference accessors here to modify those: 
```suggestion
    real64 & averagePressure = regionStatistics.getaveragePressure();
```

And further:
```c++
      regionStatistics.getPhaseMass(ip) += subRegionPhaseMass[ip];
```

This way, not more need for those public `RegionStatistics::viewKeyStruct`.

If you must stick to using those intermediate wrappers, I think it is better to use the `statsVKS` to prevent long lines, but try not to.

And also for `SinglePhaseStatistics`.